microservice/beamable.tooling.common/Microservice/ServiceMethodHelper.cs

```diff
@@ -269,6 +269,14 @@ private static List<ServiceMethod> ScanType(MicroserviceAttribute serviceAttribu
             var requiredUser = attribute.RequireAuthenticatedUser;
 
             Log.Debug("Found {method} for {path}", method.Name, servicePath);
+            
+            var isAsync = null != method.GetCustomAttribute<AsyncStateMachineAttribute>();
+            var isVoid = method.ReturnType == typeof(void);
+            if (isAsync && isVoid)
+            {
+	            throw new BeamableMicroserviceException($"The following method is invalid, method=[{type.Name}.{method.Name}]. Callable methods in Beamable Microservices are not allowed to have the `async void` method signature. Consider using `async Promise` or `async Task` instead. ");

```

Comment:
When does this get thrown? Upon compilation? Upon deployment? Upon first invocation with a client? Any of these is acceptable, but I just wanna know so I can best guide people who are using microservices.

Reply:
it'll chuck when the user _starts_ the microservice. However, our publish flow starts the microservice as a health-cehck pass, which means the start method gets called, so publish will flunk.